File: src/leetcode/2026/05/Solution925.java
public class Solution925 {
    public boolean isLongPressedName(String name, String typed) {
        int i = 0; // Pointer for name
        int j = 0; // Pointer for typed

        while (i < name.length() && j < typed.length()) {
            if (name.charAt(i) == typed.charAt(j)) {
                i++;
                j++;
            } else if (j > 0 && typed.charAt(j) == typed.charAt(j - 1)) {
                j++;
            } else {
                return false;
            }
        }

        if (j != typed.length()) {
            while (j < typed.length()) {
                if (typed.charAt(j) != typed.charAt(j - 1)) {
                    return false;
                }
                j++;
            }
        }

        // Check if all characters in name were matched
        return i == name.length();
    }

    public static void main(String[] args) {
        Solution925 solution = new Solution925();

        String name1 = "alex";
        String typed1 = "aaleex";
        boolean result1 = solution.isLongPressedName(name1, typed1);
        System.out.println("Output: " + result1); // Output: true

        String name2 = "saeed";
        String typed2 = "ssaaedd";
        boolean result2 = solution.isLongPressedName(name2, typed2);
        System.out.println("Output: " + result2); // Output: false

        String name3 = "alex";
        String typed3 = "aaleexa";
        boolean result3 = solution.isLongPressedName(name3, typed3);
        System.out.println("Output: " + result3); // Output: false
    }
}

File: src/leetcode/2026/05/Solution925.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution925 {
public:
    bool isLongPressedName(string name, string typed) {
        int i = 0; // Pointer for name
        int j = 0; // Pointer for typed

        while (i < name.length() && j < typed.length()) {
            if (name[i] == typed[j]) {
                i++;
                j++;
            } else if (j > 0 && typed[j] == typed[j - 1]) {
                j++;
            } else {
                return false;
            }
        }

        if (j != typed.length()) {
            while (j < typed.length()) {
                if (typed[j] != typed[j - 1]) {
                    return false;
                }
                j++;
            }
        }

        // Check if all characters in name were matched
        return i == name.length();
    }
};

int main() {
    Solution925 solution;
    string name1 = "alex";
    string typed1 = "aaleex";
    cout << boolalpha << solution.isLongPressedName(name1, typed1) << endl; // true

    string name2 = "saeed";
    string typed2 = "ssaaedd";
    cout << boolalpha << solution.isLongPressedName(name2, typed2) << endl; // false

    string name3 = "leelee";
    string typed3 = "lleeelee";
    cout << boolalpha << solution.isLongPressedName(name3, typed3) << endl; // true

    string name4 = "alex";
    string typed4 = "aaleexa";
    cout << boolalpha << solution.isLongPressedName(name4, typed4) << endl; // false

    return 0;
}